common/courses/states.ts

```diff
@@ -12,6 +21,39 @@ export async function hasStarted(subcourse: Subcourse) {
     return startedLectures === 0;
 }
 
+
+export async function subcourseOverGracePeriod(subcourse: Subcourse) {
+    const lastLecture = await prisma.lecture.findFirst({
+        where: { subcourseId: subcourse.id },
+        orderBy: { start: 'desc' },
+    });
+
+    if (!lastLecture) {
+        return false;
+    }
+
+    const thirtyDaysAgo = new Date();
+    thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30);
+
+    return lastLecture.start < thirtyDaysAgo;
+}
+
+export async function subcourseOver(subcourse: Subcourse) {
+    const lastLecture = await prisma.lecture.findFirst({
+        where: { subcourseId: subcourse.id },
+        orderBy: { start: 'desc' },
+    });
```

Comment:
Maybe this could be moved to common/courses/lectures ?
There is already a `getFirstLecture`.